x-pack/filebeat/input/s3/input.go

```diff
@@ -0,0 +1,327 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package s3
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+	"sync"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/defaults"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/s3iface"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/harvester"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	// input name
+	inputName = "s3"
+	// MaxNumberOfMessage at one poll
+	MaxNumberOfMessage int64 = 10
+	// WaitTimeSecond for each poll
+	WaitTimeSecond int64 = 20
+)
+
+type s3Info struct {
+	name string
+	key  string
+}
+
+func init() {
+	err := input.Register(inputName, NewInput)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Input is a input for s3
+type Input struct {
+	started  bool
+	outlet   channel.Outleter
+	config   config
+	cfg      *common.Config
+	registry *harvester.Registry
+	logger   *logp.Logger
+}
+
+// NewInput creates a new s3 input
+func NewInput(cfg *common.Config, outletFactory channel.Connector, context input.Context) (input.Input, error) {
+	cfgwarn.Beta("s3 input type is used")
+
+	logger := logp.NewLogger(inputName)
+
+	config := defaultConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, errors.Wrap(err, "failed unpacking config")
+	}
+
+	awsConfig := defaults.Config()
+	awsCredentials := awssdk.Credentials{
+		AccessKeyID:     config.AccessKeyID,
+		SecretAccessKey: config.SecretAccessKey,
+	}
+
+	if config.SessionToken != "" {
+		awsCredentials.SessionToken = config.SessionToken
+	}
+
+	awsConfig.Credentials = awssdk.StaticCredentialsProvider{
```

Comment:
this never gets used, you recreate it in `Run`

Reply:
yeah I just realized I copy pasted twice 😂 deleting them! Thanks!